csharp/extractor/Semmle.Extraction/Extractor/Extractor.cs

```diff
@@ -102,7 +103,23 @@ public void MissingNamespace(string fqdn, bool fromSource)
 
         public ILogger Logger { get; private set; }
 
-        public static string Version => $"{ThisAssembly.Git.BaseTag} ({ThisAssembly.Git.Sha})";
+        public static string Version
+        {
+            get
+            {
+                // the resources for git information are always attached to the entry` assembly by our build system
+                var assembly = Assembly.GetEntryAssembly();
+                var describeAllStream = assembly.GetManifestResourceStream("git-ql-describe-all.log");
+                var headSHAStream = assembly.GetManifestResourceStream("git-ql-rev-parse.log");
```

Comment:
How and when are these files written and added to the assembly? Wouldn't it make sense to write this information into an `AssemblyInformationalVersionAttribute` instead? (And read the attribute value during runtime in this property?)

Reply:
These files are added to the assembly here: https://github.com/github/codeql/pull/16519/files#diff-7e26445df053500cd01ec7edf4e3a8886b3ce590878d7868d5e12fa8f2641d61R44

This is a "magic" bazel target that uses the `--workspace_status_command` mechanism to generate a target that pulls in these files, that we then add to the resources. Can you point me at documentation for that attribute thing? Although at build time, we only have these files available in the build system, not (easily) the strings inside those files.

Reply:
`AssemblyInformationalVersionAttribute` is a standard .NET attribute that can target assemblies: https://learn.microsoft.com/en-us/dotnet/api/system.reflection.assemblyinformationalversionattribute?view=net-8.0

It stores a string as an `InformationalVersion`. As an example, the dotnet SDK is shipping with assemblies that have the following `AssemblyInformationalVersion` in version 8.0.1, where the SHA is pointing to a `dotnet/runtime` [commit](https://github.com/dotnet/runtime/commit/bf5e279d9239bfef5bb1b8d6212f1b971c434606).
```
[assembly: AssemblyInformationalVersion("8.0.1+bf5e279d9239bfef5bb1b8d6212f1b971c434606")]
```

To include this in the assemblies, we'd either need to add an additional source file to the compilation, or we could add/overwrite the version information in `Directory.Build.props`, which is added in https://github.com/github/codeql/pull/16521.

Reply:
If this can be done with the contents of a file, then it should be technically possible whilst keeping the `--workspace_status_command` based implementation (we can use a `genrule` using those `.log` files as input).

Reply:
I don't think the bazel-based build uses the `Directory.Build.props` file, as it's not using msbuild, but generating a C# source file with a `genrule` based on the files here should be possible. I'll give that a shot, thanks for the pointers!

Reply:
I implemented something based on the attribute, imo that's a simpler/clearer solution, as it encapsulates the bazel part nicely, away from the C# extractor itself.

Reply:
The new approach looks great to me.